src/rust/engine/options/src/lib.rs

```diff
@@ -286,9 +287,15 @@ impl OptionParser {
         Ok(OptionParser { sources })
     }
 
-    pub fn parse_bool(&self, id: &OptionId, default: bool) -> Result<OptionValue<bool>, String> {
+    #[allow(clippy::type_complexity)]
+    fn parse_scalar<T: ToOwned + ?Sized>(
```

Comment:
Previously this logic was repeated across the scalar types and string. Now it's broken out into this generic helper.